internal/service/metric/influxdb/influxdb.go

```diff
@@ -0,0 +1,130 @@
+package influxdb
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	_ "github.com/influxdata/influxdb1-client" // needed due to go mod bug
+	influxdbv2 "github.com/influxdata/influxdb1-client/v2"
+
+	"github.com/slok/grafterm/internal/model"
+	"github.com/slok/grafterm/internal/service/metric"
+)
+
+// ConfigGatherer is the configuration of the InfluxDB gatherer.
+type ConfigGatherer struct {
+	Addr     string
+	Database string
+	Client   influxdbv2.Client
+}
+
+func (c *ConfigGatherer) defaults() error {
+	var err error
+	if c.Client == nil {
+		c.Client, err = influxdbv2.NewHTTPClient(
+			influxdbv2.HTTPConfig{
+				Addr: c.Addr,
+			})
+	}
+	return err
+}
+
+type gatherer struct {
+	cli influxdbv2.Client
+	cfg ConfigGatherer
+}
+
+// NewGatherer returns a new metric gatherer for influxdb backends.
+func NewGatherer(cfg ConfigGatherer) (metric.Gatherer, error) {
+	err := cfg.defaults()
+	if err != nil {
+		return &gatherer{}, nil
+	}
+
+	if cfg.Database == "" {
```

Comment:
Could you move this to the `defaults` so everything is defaulted and checked in the same place? thanks.